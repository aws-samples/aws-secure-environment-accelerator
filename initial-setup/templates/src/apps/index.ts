export * from './shared';
export * from './phase-0';
export * from './phase-1';
export * from './phase-2';
export * from './phase-3';
export * from './phase-4';
export * from './phase-5';
