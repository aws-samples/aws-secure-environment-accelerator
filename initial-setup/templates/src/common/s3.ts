import * as cdk from '@aws-cdk/core';
import { BucketProps, CfnBucket, CfnBucketProps } from '@aws-cdk/aws-s3';

export class S3 extends cdk.Construct {
  readonly s3BucketId: string;

  constructor(parent: cdk.Construct, name: string, props: BucketProps) {
    super(parent, name);

    const s3BucketName = props.bucketName ? props.bucketName : '';

    // TODO:
    const s3Bucket = new CfnBucket(this, s3BucketName, {
      bucketName: s3BucketName,
    });

    this.s3BucketId = s3Bucket.ref;
  }
}
