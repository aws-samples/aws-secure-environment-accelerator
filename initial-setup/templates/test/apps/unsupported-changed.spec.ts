import 'jest';
import * as cdk from '@aws-cdk/core';
import { S3 } from '@aws-pbmm/common-lambda/lib/aws/s3';
import { STS } from '@aws-pbmm/common-lambda/lib/aws/sts';
import { resourcesToList, stackToCloudFormation, ResourceWithLogicalId, ResourceProperties } from '../jest';
import { deployPhases } from './unsupported-changes.mocks';
import { phases } from '../../src/app';

type ResourcePropertySelector = (resource: ResourceWithLogicalId) => Partial<ResourceWithLogicalId>;

type UnsupportedChangeEntry = [string, ResourcePropertySelector];

/**
 * Returns a ResourceMatcher that matches all properties in a resource.
 */
const matchAllResourceProperties: ResourcePropertySelector = resource => ({
  LogicalId: resource.LogicalId,
  Properties: resource.Properties,
});

/**
 * Returns a ResourceMatcher for the given property names.
 */
function matchResourceProperties(propertyNames: string[]): ResourcePropertySelector {
  return resource => ({
    LogicalId: resource.LogicalId,
    Properties: selectKeys(resource.Properties, propertyNames),
  });
}

/**
 * Returns an object containing all the given properties of the given properties object.
 */
function selectKeys(properties: ResourceProperties, propertyNames: string[]): ResourceProperties {
  return propertyNames.reduce((result, propertyName) => {
    return {
      ...result,
      [propertyName]: properties[propertyName],
    };
  }, {});
}

/**
 * List of unsupported changes per resource type.
 */
const UNSUPPORTED_CHANGES: UnsupportedChangeEntry[] = [
  ['AWS::S3::Bucket', matchResourceProperties(['BucketName'])],
  ['AWS::DirectoryService::MicrosoftAD', matchAllResourceProperties],
  ['AWS::SecretsManager::Secret', matchResourceProperties(['Name'])],
  [
    'AWS::EC2::Instance',
    matchResourceProperties([
      'AvailabilityZone',
      'CpuOptions',
      'ElasticGpuSpecifications',
      'ElasticInferenceAccelerators',
      'HibernationOptions',
      'HostResourceGroupArn',
      'ImageId',
      'InstanceType',
      'Ipv6AddressCount',
      'Ipv6Addresses',
      'KeyName',
      'LicenseSpecifications',
      'NetworkInterfaces',
      'PlacementGroupName',
      'PrivateIpAddress',
      'SecurityGroups',
      'SubnetId',
    ]),
  ],
];

const stackResources: { [stackName: string]: ResourceWithLogicalId[] } = {};

beforeAll(async () => {
  // Mock STS and S3 as the IAM logic is currently contacting S3 using the SDK
  // @ts-ignore
  jest.spyOn(STS.prototype, 'getCredentialsForAccountAndRole').mockImplementation(() => Promise.resolve(undefined));
  jest.spyOn(S3.prototype, 'getObjectBodyAsString').mockImplementation(() =>
    Promise.resolve(
      JSON.stringify({
        Version: '2012-10-17',
        Statement: [
          {
            Effect: 'Allow',
            Action: '*',
            Resource: '*',
          },
        ],
      }),
    ),
  );

  const app = new cdk.App();

  // Deploy all phases that are defined in src/app.ts
  await deployPhases(app, phases);

  // Convert the stacks to CloudFormation resources
  const stacks = app.node.children.filter(child => child instanceof cdk.Stack) as cdk.Stack[];
  for (const stack of stacks) {
    const template = stackToCloudFormation(stack);
    const resources = resourcesToList(template.Resources);

    stackResources[stack.stackName] = resources;
  }
});

test.each(UNSUPPORTED_CHANGES)(
  'there should not be any unsupported resource changes for %s',
  (resourceType, resourcePropertySelector) => {
    for (const [stackName, resources] of Object.entries(stackResources)) {
      // Find all resources with the given type
      const resourcesOfType = resources.filter(r => r.Type === resourceType);
      // Select the relevant properties
      const expected = resourcesOfType.map(resourcePropertySelector);

      // Compare the relevant properties to the snapshot
      expect(expected).toMatchSnapshot(stackName);
    }
  },
);
