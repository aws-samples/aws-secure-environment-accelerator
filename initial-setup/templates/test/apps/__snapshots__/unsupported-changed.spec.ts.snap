// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-LogArchive-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-LogArchive-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "DefaultCoreBudget2890563D",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Master-Phase0 1`] = `
Array [
  Object {
    "LogicalId": "OrganizationBudgetD31584F9",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Master-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Operations-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "DefaultCoreBudget2890563D",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Perimeter-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "PerimeterBudgetA01BBAF8",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-Security-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "DefaultCoreBudget2890563D",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-SharedNetwork-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "DefaultCoreBudget2890563D",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::Budgets::Budget: PBMMAccel-SharedServices-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "DefaultCoreBudget2890563D",
    "Properties": Object {
      "NotificationsWithSubscribers": Array [
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 50,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 75,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 90,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
        Object {
          "Notification": Object {
            "ComparisonOperator": "GREATER_THAN",
            "NotificationType": "ACTUAL",
            "Threshold": 100,
            "ThresholdType": "PERCENTAGE",
          },
          "Subscribers": Array [
            Object {
              "Address": "myemail+pbmmT-budg@example.com",
              "SubscriptionType": "EMAIL",
            },
          ],
        },
      ],
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-LogArchive-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-LogArchive-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Master-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Master-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Operations-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Perimeter-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-Security-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-SharedNetwork-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::DirectoryService::MicrosoftAD: PBMMAccel-SharedServices-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-LogArchive-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-LogArchive-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Master-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Master-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Operations-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Perimeter-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-Security-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-SharedNetwork-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::EC2::Instance: PBMMAccel-SharedServices-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-LogArchive-Phase0 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
  Object {
    "LogicalId": "aescacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-LogArchive-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Master-Phase0 1`] = `
Array [
  Object {
    "LogicalId": "configcacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Master-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Operations-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Perimeter-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-Security-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-SharedNetwork-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::S3::Bucket: PBMMAccel-SharedServices-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "cacentral1",
    "Properties": Object {
      "BucketName": undefined,
      "ObjectLockEnabled": undefined,
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-LogArchive-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-LogArchive-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Master-Phase0 1`] = `
Array [
  Object {
    "LogicalId": "SecretsbgUser1UserPswdPolicyC9835C06",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsbgUser1UserPswdF10CD199",
      },
    },
  },
  Object {
    "LogicalId": "SecretsbgUser2UserPswdPolicy2131F530",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsbgUser2UserPswd08D94AA3",
      },
    },
  },
  Object {
    "LogicalId": "SecretsOpsUser1UserPswdPolicy903155AD",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsOpsUser1UserPswdD5EAD264",
      },
    },
  },
  Object {
    "LogicalId": "SecretsOpsUser2UserPswdPolicyAD9B103E",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsOpsUser2UserPswdD05A1573",
      },
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordPolicy73965873",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsMadPassword5CC78627",
      },
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordadconnectorusrPolicy889AF319",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsMadPasswordadconnectorusrB8347327",
      },
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordUser1Policy18726F39",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsMadPasswordUser1291E3AF7",
      },
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordUser2PolicyEC783CD1",
    "Properties": Object {
      "SecretId": Object {
        "Ref": "SecretsMadPasswordUser29026EDF3",
      },
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Master-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Operations-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Perimeter-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-Security-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-SharedNetwork-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::ResourcePolicy: PBMMAccel-SharedServices-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-LogArchive-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-LogArchive-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Master-Phase0 1`] = `
Array [
  Object {
    "LogicalId": "SecretsbgUser1UserPswdF10CD199",
    "Properties": Object {
      "Name": "PBMMAccel/master/user/password/bgUser1",
    },
  },
  Object {
    "LogicalId": "SecretsbgUser2UserPswd08D94AA3",
    "Properties": Object {
      "Name": "PBMMAccel/master/user/password/bgUser2",
    },
  },
  Object {
    "LogicalId": "SecretsOpsUser1UserPswdD5EAD264",
    "Properties": Object {
      "Name": "PBMMAccel/master/user/password/OpsUser1",
    },
  },
  Object {
    "LogicalId": "SecretsOpsUser2UserPswdD05A1573",
    "Properties": Object {
      "Name": "PBMMAccel/master/user/password/OpsUser2",
    },
  },
  Object {
    "LogicalId": "SecretsMadPassword5CC78627",
    "Properties": Object {
      "Name": "PBMMAccel/operations/mad/password",
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordadconnectorusrB8347327",
    "Properties": Object {
      "Name": "PBMMAccel/operations/mad/adconnector-usr/password",
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordUser1291E3AF7",
    "Properties": Object {
      "Name": "PBMMAccel/operations/mad/User1/password",
    },
  },
  Object {
    "LogicalId": "SecretsMadPasswordUser29026EDF3",
    "Properties": Object {
      "Name": "PBMMAccel/operations/mad/User2/password",
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Master-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Operations-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Operations-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Perimeter-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Perimeter-Phase1 1`] = `
Array [
  Object {
    "LogicalId": "CertPerimSelfSignedCertSecret",
    "Properties": Object {
      "Name": "accelerator/certificates/PerimSelf-SignedCert",
    },
  },
]
`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Security-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-Security-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-SharedNetwork-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-SharedNetwork-Phase1 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-SharedServices-Phase0 1`] = `Array []`;

exports[`there should not be any unsupported resource changes for AWS::SecretsManager::Secret: PBMMAccel-SharedServices-Phase1 1`] = `Array []`;
